code/lib/manager-api/src/modules/refs.ts

```diff
@@ -273,29 +277,33 @@ export const init: ModuleFn<SubAPI, SubState> = (
     },
 
     setRef: (id, { storyIndex, setStoriesData, ...rest }, ready = false) => {
+      const internalIndex = { storyIndex, setStoriesData };
```

Comment:
I think we should try and keep the same structure for refs and the SB's store; ie. `internal_index` should have the same type; ie. you should convert it to an *actual* index before storing + filtering it.

Reply:
@tmeasday I wish, but then how do I know what to call `setRef:` with?

We have no way to convert between `storyIndex` and `setStoriesData`?

I also didn't want to make a lot of changes considering if we're interested in publishing for the addon, we need to do so in a patch version ,and I'd want to make a few changes as possible to minimize risk.

Reply:
@ndelangen 

> We have no way to convert between storyIndex and setStoriesData?

https://github.com/storybookjs/storybook/blob/04ced719c25b81af8ab11813acf54f856b8fe3f8/code/lib/manager-api/src/lib/stories.ts#L52-L55

> what to call setRef: with

Couldn't you just always pass the index? I'm not sure.

Reply:
No because then neither `transformSetStoriesStoryDataToStoriesHash` nor `transformStoryIndexToStoriesHash` will be called, which is where the filtering happens.

Reply:
Can't we do something like:

```ts
setRef(id, { storyIndex, setStoriesData, ...rest }) {

  const internal_index = setStoriesData ? transformSetStoriesStoryDataToPreparedStoryIndex(setStoriesData) : storyIndex;

  const index = transformStoryIndexToStoriesHash(storyIndex, { filters })

  api.updateRef(id, { ...rest, index, internal_index });
}
```